langchain4j-core/src/main/java/dev/langchain4j/model/chat/response/ChatResponse.java

```diff
@@ -7,21 +7,34 @@
 
 import java.util.Objects;
 
+import static dev.langchain4j.internal.Utils.quoted;
 import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
 
 @Experimental
 public class ChatResponse {
 
+    private final String id;
+    private final String modelName; // TODO name
     private final AiMessage aiMessage;
     private final TokenUsage tokenUsage;
     private final FinishReason finishReason;
 
-    private ChatResponse(Builder builder) {
+    protected ChatResponse(Builder builder) { // TODO
```

Comment:
Same thing as for the `ChatRequest`, this could also become a `record`.

Reply:
Right, I am still adapting to java 17 baseline :)

Reply:
As far as I understand, converting `ChatResponse` into a record will require exposing a public constructor that accepts all fields. I wanted to keep the only constructor private: `private ChatRequest(Builder)`, to have the flexibility to add more fields to the `ChatResponse` in the future without any breaking changes and the need to introduce more overloaded ctors. Is this a valid approach?

Reply:
You're right, I wasn't thinking that turning this class into a record will force you to have a public constructor that clearly you don't want to expose. The only alternative that I see, in case that you will decide to introduce an interface as we were discussing for the `ChatRequest`, is that at that point you could keep the whole record (the default implementation of that interface) as package private thus completely avoiding to expose it in your public API.

Reply:
I was thinking to implement the request and response in the same way purely for symmetry, but you're right this doesn't make much sense. What I wrote above only applies to the `ChatRequest` then.

Reply:
However if I understand correctly, users won't have any reason also to call that `ChatResponse` constructor (they never create that object) even if would be publicly available, right?

Reply:
I guess they might call it when testing?
When they will need to mock `ChatModel.chat(ChatRequest)` method, they will need to create a `ChatResponse` object somehow.

AFAIU, if something is public (even if in theory it should not be used by anyone), it will be used anyway for some reason (good or bad 😆), so we need to treat it as public and avoid breaking changes. WDYT?

Reply:
Agreed, better to play safe with public API ... and you never know what people actually could do with it :smile: